### PR TITLE
Move module development guidelines into the collection

### PR DESCRIPTION
**Pull request by tremble**
##### SUMMARY

As per https://github.com/ansible/ansible/pull/77598#issuecomment-1105416249
```
@samccann:
This guide should be moved to one or the other of the AWS collections (in the docs/docsite/rst folder like the scenario guide is already).
```

##### ISSUE TYPE

- Docs Pull Request

##### COMPONENT NAME

docs/docsite/

### CODE REVIEW

File: docs/docsite/rst/dev_guidelines.rst [hunk 1]
```diff
@@ -0,0 +1,805 @@
...
+
+.. _ansible_collections.amazon.aws.docsite.dev_module_maint:
+
+Maintaining existing modules
+============================
+
+Fixing bugs
+-----------
+
+Bug fixes to code that relies on boto will still be accepted. When possible,
```
Comment (alinabuzachis) [inline docs/docsite/rst/dev_guidelines.rst:62]: Do we still have modules using boto instead of boto3?

Comment (tremble) [reply docs/docsite/rst/dev_guidelines.rst:62]: No.  I have a separate PR https://github.com/ansible/ansible/pull/77598 which cleans this up, but I wanted to start with the minimum necessary change, then layer the extra changes on top.

**Comment by samccann** [comment]
@tremble thanks for moving this over!  I have some scribble-notes below on how to test these docs locally if you want to. Otherwise we can see what they look like once they are in devel docs.

There is currently no easy way to verify guides within a collection will build properly. For the interim, the [community.hashi_vault](https://docs.ansible.com/ansible/devel/collections/community/hashi_vault/docsite/contributor_guide.html#collection-docsite) collection is experimenting with docs build script. To use this locally:

1. Create a clean venv for this testing and pip install ansible.
2. Copy the build.sh, conf.py, and requirements.txt from the [community.hashivault preview directory](https://github.com/ansible-collections/community.hashi_vault/tree/main/docs/preview) into your own docs/preview directory for your test collection.
3. Edit [build.sh](https://github.com/ansible-collections/community.hashi_vault/blob/main/docs/preview/build.sh#L11) to change the name to your test collection.
4. Install the requirements in your new venv: `pip install -r requirements.txt.`
5. Add your rst files and extra-docs.yml as explained [here](https://docs.ansible.com/ansible/latest/dev_guide/developing_collections_structure.html#docs-directory).
6. Run the script to include the directory where your test collection lives as ansible_collections/my/collection : ANSIBLE_COLLECTIONS=~/collections-sources/ ./build.sh
7. Review the output and fix any problems.

**Review by samccann — APPROVED** [review]
LGTM

**Review by felixfontein — APPROVED** [review]
I ran `antsibull-docs lint-collection-docs` on this PR, it didn't report problems.

**Comment by felixfontein** [comment]
> * Copy the build.sh, conf.py, and requirements.txt from the [community.hashivault preview directory](https://github.com/ansible-collections/community.hashi_vault/tree/main/docs/preview) into your own docs/preview directory for your test collection.

You can also use `antsibull-docs sphinx-init` to create it directly :)

It's also possible to not use any of this, but include something like https://github.com/ansible-collections/community.docker/blob/main/.github/workflows/docs-pr.yml that shows a docs diff for every PR (and thus also does a docs build for every PR).

**Review by jillr — APPROVED** [review]

**Comment by felixfontein** [comment]
Out of curiosity, is there something in CI that runs `antsibull-docs lint-collection-docs`?

**Comment by tremble** [comment]
Since the changes for this are so minimal and things have been manually tested, I'm going to mergeit and backport to 3.x.  With #790 I'll try to get @felixfontein's suggestion to work

**Comment by briantist** [comment]
> @tremble thanks for moving this over! I have some scribble-notes below on how to test these docs locally if you want to. Otherwise we can see what they look like once they are in devel docs.
> 
> There is currently no easy way to verify guides within a collection will build properly. For the interim, the [community.hashi_vault](https://docs.ansible.com/ansible/devel/collections/community/hashi_vault/docsite/contributor_guide.html#collection-docsite) collection is experimenting with docs build script. To use this locally:
> 
> 1. Create a clean venv for this testing and pip install ansible.
> 2. Copy the build.sh, conf.py, and requirements.txt from the [community.hashivault preview directory](https://github.com/ansible-collections/community.hashi_vault/tree/main/docs/preview) into your own docs/preview directory for your test collection.
> 3. Edit [build.sh](https://github.com/ansible-collections/community.hashi_vault/blob/main/docs/preview/build.sh#L11) to change the name to your test collection.
> 4. Install the requirements in your new venv: `pip install -r requirements.txt.`
> 5. Add your rst files and extra-docs.yml as explained [here](https://docs.ansible.com/ansible/latest/dev_guide/developing_collections_structure.html#docs-directory).
> 6. Run the script to include the directory where your test collection lives as ansible_collections/my/collection : ANSIBLE_COLLECTIONS=~/collections-sources/ ./build.sh
> 7. Review the output and fix any problems.

@samccann Thanks for mentioning! I would recommend pointing people to the GitHub actions for this; these can be used today to verify a clean docs build in collections: https://github.com/ansible-community/github-docs-build

(I know I still have docs to write on using it; for now I can still help on a collection by collection basis)

The `build.sh` script above doesn't need to be copied from my collection, it's probably best generated by `antsibull-docs sphinx-init`.

**Comment by samccann** [comment]
@briantist - I tried to put the steps in [this hackmd note](https://hackmd.io/w6JSHW_XRzi9cBZtoN4snA). Can you edit/correct anything I got wrong? I'm not familiar with github actions so that area is.. ahem.. vague :-P.

**Comment by briantist** [comment]
@samccann I made some edits, let me know if they are confusing/unhelpful. We can chat in IRC tomorrow too.

**Pull request merged**